app/src/common/shared/org/mozilla/vrbrowser/telemetry/GleanMetricsService.java

```diff
@@ -0,0 +1,60 @@
+package org.mozilla.vrbrowser.telemetry;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.UiThread;
+
+import org.mozilla.vrbrowser.browser.SettingsStore;
+import org.mozilla.vrbrowser.utils.DeviceType;
+import org.mozilla.vrbrowser.utils.SystemUtils;
+import org.mozilla.vrbrowser.BuildConfig;
+import org.mozilla.vrbrowser.GleanMetrics.Distribution;
+
+import mozilla.components.service.glean.Glean;
+import mozilla.components.service.glean.config.Configuration;
+
+
+public class GleanMetricsService {
+
+    private final static String APP_NAME = "FirefoxReality";
+    private static boolean initialized = false;
+    private final static String LOGTAG = SystemUtils.createLogtag(GleanMetricsService.class);
+    private static Context context = null;
+
+    // We should call this at the application initial stage.
+    public static void init(Context aContext) {
+        if (initialized)
+            return;
+
+        context = aContext;
+        initialized = true;
+
+        final boolean telemetryEnabled = SettingsStore.getInstance(aContext).isTelemetryEnabled();
+        if (telemetryEnabled) {
+            GleanMetricsService.start();
```

Comment:
@Dexterp37 I did a little adjustment here because I aware it would be very easy to forget to call `setStartupMetrics()` if we only call `Glean.INSTANCE.setUploadEnabled(true);` here.